[Resubmit] Require inductor to match stride order

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91185

Resubmitting https://github.com/pytorch/pytorch/pull/90563 because I had a commit in that stack which didn't use my CLA-approved git username. 


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

eellison committed:
[Resubmit] Require inductor to match stride order

[ghstack-poisoned]

desertfire commented on test/inductor/test_torchinductor_opinfo.py:
```diff
@@ -273,22 +274,22 @@ def process(device_type):
     "cov": {f16, f32, f64, i32, i64},
     "equal": {b8, f16, f32, f64, i32, i64},
-    "fft.fft": {f16, f32, f64},
+    "fft.fft": {b8, f16, f32, f64, i32, i64},
```
What's the story about these new expected failures?

eellison replied:
fft support here is extremely flakey. these kernels output complex, so the `requiring_stride_order` on the end breaks when we interact with complex dtypes

eellison committed:
Update on "[Resubmit] Require inductor to match stride order"


Resubmitting https://github.com/pytorch/pytorch/pull/90563 because I had a commit in that stack which didn't use my CLA-approved git username. 


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

anijain2305 commented on torch/_inductor/graph.py:
```diff
@@ -366,6 +366,20 @@ def run_node(self, n: torch.fx.Node):
+            # require the same stride order for dense outputs,
+            # so that user-land view() will not throw because inductor
+            # output different strides than eager
+            # long term the solution is to make view() always succeed
+            # with infallible strides.
```
Just for understand (the PR is ok) - What is view with infallible strides?

eellison replied:
https://docs.google.com/document/d/1wWUd_hBoCwnMV2CnJNWxNFMRZYFyspV_iyI_1Nya7c0/edit#heading=h.lmmok2ikj8ad

eellison committed:
Update on "[Resubmit] Require inductor to match stride order"


Resubmitting https://github.com/pytorch/pytorch/pull/90563 because I had a commit in that stack which didn't use my CLA-approved git username. 


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison commented:
@pytorchbot rebase

eellison committed:
Update on "[Resubmit] Require inductor to match stride order"


Resubmitting https://github.com/pytorch/pytorch/pull/90563 because I had a commit in that stack which didn't use my CLA-approved git username. 


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "[Resubmit] Require inductor to match stride order"


Resubmitting https://github.com/pytorch/pytorch/pull/90563 because I had a commit in that stack which didn't use my CLA-approved git username. 


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison commented:
@pytorchbot merge -f "eager variation failures"
